File: src/types/contract.ts
export interface Contract {
  contractId?: number;
  procurementRequestId: number;
  contractName: string;
  contractCode?: string;
  supplierId?: number;
  amount?: number;
  signingDate?: string;
  attachmentUrl?: string;
  status?: string;
  
  // Extended fields for management
  signerName?: string;
  signerContact?: string;
  vendorSignerName?: string;
  vendorSignerContact?: string;
  
  // Audit fields
  createTime?: string;
  createUserName?: string;
}

export interface ContractArchivingItem {
  procurementRequestId: number;
  procurementRequestCode: string;
  procurementTitle: string;
  amount: number;
  applicantName: string;
  department: string;
  createTime: string;
  
  evaluationCompleted: boolean;
  bidId?: number;
  
  contractId?: number;
  contractStatus: string; // PENDING_ENTRY, ENTERED
  contractName?: string;
  signingDate?: string;
}

File: backend/src/main/java/com/example/procurement/entity/Contract.java
package com.example.procurement.entity;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Contract {
    private Long contractId;
    private Long procurementRequestId;
    private String contractName;
    private String contractCode;
    private Long supplierId;
    private BigDecimal amount;
    private LocalDateTime signingDate;
    private String attachmentUrl;
    private String status;
    private String signerName;
    private String signerContact;
    private String vendorSignerName;
    private String vendorSignerContact;
    private LocalDateTime createTime;
    private String createUserId;
    private String createUserName;
    private LocalDateTime updateTime;
    private String updateUserId;
    private String updateUserName;
}
